src/app/settings/keyboard-mapping-settings/components/edit-keyboard-mapping/edit-keyboard-mapping.component.ts

```diff
@@ -33,7 +33,7 @@ export class EditKeyboardMappingComponent implements OnInit {
       label: [this.keyboardMapping?.actionTrigger.data.label ?? ''],
       keys: [this.keyboardMapping?.actionTrigger.data.keys, [Validators.required]],
       triggerOn: [KeyEventType.RELEASED, [Validators.required]],
-      mappedToKeys: [this.keyboardMapping?.actionTrigger.data.mappedToKeys],
+      mappedToKeys: [this.keyboardMapping?.actionTrigger.data.mappedToKeys ?? []],
```

Comment:
Why is this added?

Reply:
Without it, the InputField for MappedKeys would display the "x" to clear the field, but the field was already "clear". So this makes the field behave correctly.

Reply:
I just tried removing this and nothing changed in the behaviour of the keyboard input field. This changes the runtime type of mappedToKeys from `Keys | undefined` to `Keys | string[] | undefined`.

Reply:
Sorry, I got it a bit messed up. If you don't have the `?? []` then when you create a new KeyboardMapping, the `mappedKeys` is going to have the value `null`, which is not allowed in our Zod parser.

Try creating a new KeyboardMapping and check your console.

Reply:
If we use `[]` as the default value for the mappedToKeys input field, then we should check for an empty array and convert it to `undefined` when saving.
Another solution is to make the default value of the KeyboardInputComponent `undefined` instead of `null`.

Reply:
The default value of KeyboardInputComponent is already `undefined`. It's Angular forms that puts it as `null` when you don't provide a value.
If you give the input field a value and then clear it with the "x", then the value is going to be `undefined`.

Reply:
The exposed default value should be `undefined` instead of `null`.

Reply:
As mentioned in my comment above. The default value in the InputField is already `undefined`. It's Angular forms that changes the value to `null` when you don't provide a value to the `FormControl`.

Reply:
We'll leave it as is and I'll create a task for fixing it.